.github/workflows/validate-migration.yaml

```diff
@@ -73,29 +171,43 @@ jobs:
           mkdir -p vendor/helix-core-api/linux
           tar -C vendor/helix-core-api/linux -xzf p4.tgz --strip 1
 
-      - name: "Cache OpenSSL"
+      - name: "Cache OpenSSL source"
+        id: cache-openssl-source
         uses: actions/cache@v2
-        id: cache-openssl
         with:
-          path: ${{ env.OPENSSL_INSTALL_DIR }}
-          key: ${{ runner.os }}-openssl-${{ env.OPENSSL_DOWNLOAD_URL }}
+          path: ${{ env.OPENSSL_SOURCE_DIR }}
+          key: ${{ runner.os }}-openssl-source-${{ env.OPENSSL_DOWNLOAD_URL }}
 
-      - name: "Compile OpenSSL"
-        if: ${{ steps.cache-openssl.outputs.cache-hit != 'true' }}
+      - name: "Download OpenSSL source"
+        if : ${{ steps.cache-openssl-source.outputs.cache-hit != 'true' }}
         run: |
-          mkdir openssl-src
           curl -L "${OPENSSL_DOWNLOAD_URL}" --output openssl.tgz
-          tar -C openssl-src -xzf openssl.tgz --strip 1
-          pushd openssl-src
-          ./config --prefix="${OPENSSL_INSTALL_DIR}" --static
-          make install
-          popd 
+          mkdir -p "${OPENSSL_SOURCE_DIR}" || true
+          tar -C "${OPENSSL_SOURCE_DIR}" -xzf openssl.tgz --strip-components 1
+          rm openssl.tgz || true
+
+      - name: "Compile OpenSSL"
+        run: |
+          .github/workflows/validate-migration-scripts/compile-openssl.sh
+        env:
+          CC: "sccache clang"
+          CXX: "sccache clang++"
 
       - name: "Run validate migration script"
         run: |
-          sudo apt-get install gettext --yes # for envsubst 
-          .github/workflows/run-test.sh
+          .github/workflows/validate-migration-scripts/run-test.sh
```

Comment:
Maybe this script and YAML step need to be renamed? The usage of "run" makes it sound like you're just going to run the built code, but this step is doing both building and running. So maybe `Build p4-fusion and test migration validation`?